docs/guide/examples.rst

```diff
@@ -491,6 +491,98 @@ Behind the scene, SB3 uses an :ref:`EvalCallback <callbacks>`.
 
 
 
+Accessing and modifying model parameters
+----------------------------------------
+
+You can access model's parameters via ``load_parameters`` and ``get_parameters`` functions,
+or via ``model.policy.state_dict()`` (and ``load_state_dict()``),
+which use dictionaries that map variable names to PyTorch tensors.
+
+These functions are useful when you need to e.g. evaluate large set of models with same network structure,
+visualize different layers of the network or modify parameters manually.
+
+Policies also offers a simple way to save/load weights as a NumPy vector, using ``parameters_to_vector()``
+and ``load_from_vector()`` method.
+
+Following example demonstrates reading parameters, modifying some of them and loading them to model
+by implementing `evolution strategy (es) <http://blog.otoro.net/2017/10/29/visual-evolution-strategies/>`_
+for solving ``CartPole-v1`` environment. The initial guess for parameters is obtained by running
+A2C policy gradient updates on the model.
+
+.. code-block:: python
+
+  from typing import Dict
+
+  import gym
+  import numpy as np
+  import torch as th
+
+  from stable_baselines3 import A2C
+  from stable_baselines3.common.evaluation import evaluate_policy
+
+
+  def mutate(params: Dict[str, th.Tensor]) -> Dict[str, th.Tensor]:
+      """Mutate parameters by adding normal noise to them"""
+      return dict((name, param + th.randn_like(param)) for name, param in params.items())
+
+
+  # Create policy with a small network
+  model = A2C(
+      "MlpPolicy",
+      "CartPole-v1",
+      ent_coef=0.0,
+      policy_kwargs={"net_arch": [32]},
+      seed=0,
+      learning_rate=0.05,
+  )
+
+  # Use traditional actor-critic policy gradient updates to
+  # find good initial parameters
+  model.learn(total_timesteps=10000)
+
+  # Include only variables with "policy", "action" (policy) or "shared_net" (shared layers)
+  # in their name: only these ones affect the action.
+  # NOTE: you can retrieve those parameters using model.get_parameters() too
```

Comment:
Hmm we end up with many options for modifying parameters, should probably differentiate what is the right way of doing it / what are the differences. Maybe this example could use `get_parameters` as the goal is to contain (almost) all necessary parameters for value predictions etc.

Reply:
That's documented in the export section ;)
I would say that PyTorch internal method is recommended in most cases but `get_parameters()` is needed if you want everything (including optimizers).

>Maybe this example could use get_parameters as the goal is to contain (almost) all necessary parameters for value predictions etc.

in practice, it is not so clean with `get_parameters()`...

Reply:
Fair enough :). I agree torch's way is cleaner (and especially with the neat vectorization functions for policy), which we should promote first whenever possible.